osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -86,7 +86,20 @@ protected override PerformanceAttributes CreatePerformanceAttributes(ScoreInfo s
 
         private double computeAimValue(ScoreInfo score, OsuDifficultyAttributes attributes)
         {
-            double aimValue = Math.Pow(5.0 * Math.Max(1.0, attributes.AimDifficulty / 0.0675) - 4.0, 3.0) / 100000.0;
+            double aimDifficulty = attributes.AimDifficulty;
+
+            // We assume 25% of sliders in a map are difficult since there's no way to tell from the performance calculator.
+            double estimateDifficultSliders = attributes.SliderCount * 0.25;
+
+            if (attributes.SliderCount > 0)
+            {
+                // Meh and Miss are more severe on sliders than normal Ok, so punish them 2 times more
+                double estimateSliderEndsDropped = Math.Clamp(Math.Min(countOk + countMeh * 2 + countMiss * 2, attributes.MaxCombo - scoreMaxCombo), 0, estimateDifficultSliders);
+                double sliderNerfFactor = (1 - attributes.SliderFactor) * (1 - estimateSliderEndsDropped / estimateDifficultSliders) + attributes.SliderFactor;
+                aimDifficulty *= sliderNerfFactor;
+            }
+
+            double aimValue = Math.Pow(5.0 * Math.Max(1.0, aimDifficulty / 0.0675) - 4.0, 3.0) / 100000.0;
```

Comment:
I'm not really sure about this entire section. Something seems weird about adjusting the aim difficulty attributes rather than adjusting the aim value itself. It ruins the model of base values always being the same for a star rating but I might just be picky, open to other opinions.

Reply:
moving sliderend punishments to the difficulty attribute makes perfect sense from the fact what `SliderFactor` is

Reply:
I would prefer it to be cubic here.

Reply:
> I would prefer it to be cubic here.
okay, i will change it then